### ISSUE TITLE
Some issue in action view caching after updating to rails 3.2.22.2

### ISSUE THREAD

**Issue by vsorlov** [rails/rails#24159]
Hi, after updating to rails 3.2.22.2 we have an issue. I'll try to add a failing test.

### Steps to reproduce

when trying to render a file, `TemplateRenderer` receives the following options `{:file=>"public/404", :formats=>[:html], :status=>404, :template=>"index", :layout=>nil}`
because of this change https://github.com/rails/rails/compare/v3.2.22.1...v3.2.22.2#diff-d010ed2738c7b1e87f729689bbc8f295R29 it fails

### Expected behavior
no error

### Actual behavior
some requests fail with an error

```
undefined method `[]' for nil:NilClass
# .../gems/actionpack-3.2.22.2/lib/action_view/template/resolver.rb:81:in `cached'
# .../gems/actionpack-3.2.22.2/lib/action_view/template/resolver.rb:51:in `find_all_anywhere'
# .../gems/actionpack-3.2.22.2/lib/action_view/path_set.rb:80:in `block (2 levels) in _find_all'
```

### System configuration
**Rails version**: rails 3.2.22.2

**Ruby version**: ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-darwin13.0]

**Comment by rafaelfranca** [comment]
> Steps to reproduce

There are not the steps to reproduce. I can't reproduce the issue with these steps. Could you please provide steps to reproduce this issue?

**Comment by vsorlov** [comment]
@rafaelfranca, seems it is something specific to our app. I can reproduce it with the following rspec test in our app
```ruby
  1 describe ActionController::Base, type: :controller do
  2   describe '#index' do
  3     controller do
  4       def index
  5         render file: 'public/404', formats: [:html], layout: false, status: :not_found
  6       end
  7     end
  8
  9     it 'responds with 404 status' do
 10       get :index
 11       expect(response.status).to eq(404)
 12     end
 13   end
 14 end
```

but in rails unit tests this code works without an error. I'm investigating the code, but didn't find the cause of the problem yet. I think it is better to close this issue for now, I'll reopen it if I find that the cause is in rails code.

**Comment by rafaelfranca** [comment]
I think rspec had an issue with this fix when applied in Rails 4.2.

**Comment by jnixon22** [comment]
Did you find a root cause for this?  I'm seeing something similar when upgrading to 3.2.22.2.

**Comment by sevaorlov** [comment]
@jnixon22, yes, there was a fix in Rspec for it, so I just upgraded Rspec

**Comment by jnixon22** [comment]
@sevaorlov Was the fix was in Rspec 3.X somewhere?  I'm currently stuck on 2.14.2.

**Comment by sevaorlov** [comment]
@jnixon22, yes, somewhere in Rspec 3, there was some commit that fixed compatibility with Rails 4 and Rails 3.2.22.2 added some code from Rails 4, that is why that commit was needed.
